pkg/vetter/util/mtlspolicy/authPolicy_test.go

```diff
@@ -160,56 +193,475 @@ var (
+		Expect(loaded.ByMesh()).To(Equal([]*authv1alpha1.MeshPolicy{meshDefaultOn}))
+		Expect(loaded.ByNamespace("barNs")).To(Equal([]*authv1alpha1.Policy{nsbarNs_On}))
 		Expect(loaded.ByNamespace("default")).To(Equal([]*authv1alpha1.Policy{}))
 
 		Expect(loaded.ByName(foo)).To(Equal([]*authv1alpha1.Policy{
-			apFooOn,
-			apFooOff,
-			apFooBarOn,
-			// no apFooPortsBarOn because that is only for foo:8443, not foo
+			nsDefault_apFoo_On,
+			nsDefault_apFoo_Off,
+			nsDefault_apFoo_apBar_On,
+			// no nsDefault_apFooPorts_apBar_On because that is only for foo:8443, not foo
 		}))
 		Expect(loaded.ByName(bar)).To(Equal([]*authv1alpha1.Policy{
-			apFooBarOn,
-			// apFooPortsBarOn because that is for bar (not bar:8443)
-			apFooPortsBarOn,
+			nsDefault_apFoo_apBar_On,
+			// nsDefault_apFooPorts_apBar_On because that is for bar (not bar:8443)
+			nsDefault_apFooPorts_apBar_On,
 		}))
 
-		Expect(loaded.ByPort(foo, 8443)).To(Equal([]*authv1alpha1.Policy{apFooPortsBarOn}))
+		Expect(loaded.ByPort(foo, 8443)).To(Equal([]*authv1alpha1.Policy{nsDefault_apFooPorts_apBar_On}))
 		Expect(loaded.ByPort(foo, 1000)).To(Equal([]*authv1alpha1.Policy{}))
 		Expect(loaded.ByPort(bar, 8443)).To(Equal([]*authv1alpha1.Policy{}))
 	})
 })
 
-var _ = Describe("GetClusterPortPolicies", func() {
-	It("should return port policies from the loaded AuthPolicies struct", func() {
-		loaded, err := LoadAuthPolicies([]*authv1alpha1.Policy{
-			apDefaultOn,
-			apFooOn,
-			apFooOff,
-			apFooBarOn,
-			apFooPortsBarOn,
+var _ = Describe("AuthPolicyIsMtls", func() {
+	It("should evaluate no-mtls-peer as false", func() {
+		Expect(AuthPolicyIsMtls(nsDefault_apFoo_Off)).To(Equal(MTLSSetting_DISABLED))
+	})
+	It("should evaluate empty-mtls-peer as true", func() {
+		Expect(AuthPolicyIsMtls(nsDefault_apFooPorts_apBar_On)).To(Equal(MTLSSetting_ENABLED))
+	})
+})
+
+var _ = Describe("test AuthPolicyIsMtls() through its callers", func() {
+	// Context("TLSDetailsByNamespace()", func() {
+
+	// 	It("returns enabled when there is an enabled policy", func() {
+	// 		loadedOn, err := LoadAuthPolicies([]*authv1alpha1.Policy{
+	// 			nsbarNs_On,
+	// 		})
+	// 		loadedOn.LoadMeshPolicy([]*authv1alpha1.MeshPolicy{meshDefaultOn})
+
+	// 		Expect(err).To(BeNil())
+	// 		s := Service{Namespace: "barNs", Name: ""}
+	// 		mtlsStateOn, _, err1 := loadedOn.TLSDetailsByNamespace(s)
+	// 		Expect(err1).To(BeNil())
+
+	// 		Expect(mtlsStateOn).To(Equal(MTLSSetting_ENABLED))
+	// 	})
+	// 	It("returns enabled when there is no policy for a namespace, but the mesh policy exists and is enabled", func() {
+	// 		loadedNone, err := LoadAuthPolicies([]*authv1alpha1.Policy{})
+	// 		Expect(err).To(BeNil())
+	// 		loadedNone.LoadMeshPolicy([]*authv1alpha1.MeshPolicy{meshDefaultOn})
+	// 		s := Service{Namespace: "barNs", Name: ""}
+	// 		mtlsStateNone, _, err2 := loadedNone.TLSDetailsByNamespace(s)
+	// 		Expect(err2).To(BeNil())
+
+	// 		Expect(mtlsStateNone).To(Equal(MTLSSetting_ENABLED))
+	// 	})
+	// })
```

Comment:
You also need the test for "Mesh is on, namespace is off" -> DISABLED.

Once you write that one, I think you should try to refactor these tests to use a common helper since they're all very similar:

1) Load some case-specific policy and meshpolicy
2) Call TLSDetailsByNamespace("barNs")
3) Check if the result is MTLSSetting_ENABLED or disabled.